src/java/org/apache/cassandra/index/sai/utils/TermIterator.java

```diff
@@ -53,6 +52,13 @@ private TermIterator(RangeIterator union, Set<SSTableIndex> referencedIndexes, Q
         this.union = union;
         this.referencedIndexes = referencedIndexes;
         this.context = queryContext;
+
+        for (SSTableIndex index : referencedIndexes)
+        {
+            boolean success = index.reference();
```

Comment:
Why do we need a second reference if these are already referenced?

Reply:
Because TermIterator releases the reference at closing. 
This way TermIterator lifetime is decoupled from QueryView lifetime and others. 
If it didn't increase a reference count, we could end up in a situation of a dangling reference - that is, we'd have a reference to the index, but index would be already released.